src/Mod/Sketcher/Gui/Command.cpp

```diff
@@ -978,16 +977,304 @@ void CmdSketcherViewSection::activated(int iMsg)
+}
+
+/* Grid comp */
+DEF_STD_CMD_ACL(CmdSketcherCompGrid)
+
+CmdSketcherCompGrid::CmdSketcherCompGrid()
+    : Command("Sketcher_CompGrid")
+{
+    sAppModule = "Sketcher";
+    sGroup = "Sketcher";
+    sMenuText = QT_TR_NOOP("Activate Grid");
+    sToolTipText = QT_TR_NOOP("Activate grid and further settings");
+    sWhatsThis = "Sketcher_CompGrid";
+    sStatusTip = sToolTipText;
+    eType = 0;
+}
+
+void CmdSketcherCompGrid::activated(int iMsg)
+{
+    Gui::ActionGroup* pcAction = qobject_cast<Gui::ActionGroup*>(getAction());
+    if (!pcAction)
+        return;
+    QList<QAction*> a = pcAction->actions();
+
+    if (iMsg == 0) {
+        bool newState = toggleGrid(getActiveGuiDocument());
+        a[0]->setChecked(newState);
+        //activate/deactivate others
+        a[1]->setEnabled(newState);
+        a[2]->setEnabled(newState);
+    }
+    else if (iMsg == 1) {
+        toggleGridSnap(getActiveGuiDocument());
+        pcAction->setProperty("defaultAction", QVariant(0));
+    }
+    else if (iMsg == 2){
+        bool newState = toggleGridAuto(getActiveGuiDocument());
+        a[3]->setEnabled(!newState);
+        pcAction->setProperty("defaultAction", QVariant(0));
+    }
+    else if (iMsg == 3) {
+        //nothing but don't return either.
+    }
+    else
+        return;
+
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/General");
+    pcAction->setIcon(Gui::BitmapFactory().iconFromTheme(hGrp->GetBool("ShowGrid") ? "Sketcher_GridToggle" : "Sketcher_GridToggle_Deactivated"));
+}
+
+Gui::Action* CmdSketcherCompGrid::createAction()
+{
+    Gui::ActionGroup* pcAction = new Gui::ActionGroup(this, Gui::getMainWindow());
+    pcAction->setDropDownMenu(true);
+    pcAction->setExclusive(false);
+    applyCommandData(this->className(), pcAction);
+
+    QAction* a0 = pcAction->addAction(QString());
+    QAction* a1 = pcAction->addAction(QString());
+    QAction* a2 = pcAction->addAction(QString());
+
+    auto* sizeLabel = new QLabel(QString::fromLatin1("Size"));
+    auto* gridSizeBox = new Gui::QuantitySpinBox();
+    gridSizeBox->setProperty("unit", QVariant(QString::fromLatin1("mm")));
+    gridSizeBox->setObjectName(QString::fromLatin1("gridSize"));
+    gridSizeBox->setToolTip(QApplication::translate("Sketcher_GridSize", "Distance between two subsequent grid lines"));
+    //Is General/GridSize correct? From taskSketcherGeneral it seems yes.
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/General/GridSize");
+    gridSizeBox->setValue(hGrp->GetFloat("GridSize", 10.0));
+    gridSizeBox->setSingleStep(1.0);
+    gridSizeBox->setMaximum(99999999.0);
+    gridSizeBox->setMinimum(0.001);
+    gridSizeBox->setDecimals(3);
+
+    QObject::connect(gridSizeBox, static_cast<void(Gui::QuantitySpinBox::*)(double)>(&Gui::QuantitySpinBox::valueChanged),

```

Comment:
For ambiguous signals/slots you usually use qOverload:
`QObject::connect(gridSizeBox, qOverload<double>(&Gui::QuantitySpinBox::valueChanged),`